File: 1944 number-of-visible-people-in-a-queue/solution.cs
public class Solution {
    public int[] CanSeePersonsCount(int[] heights) {
        int n = heights.Length;
        int[] ans = new int[n];
        int[] st = new int[n];
        int top = -1;
        for(int i=n-1;i>=0;i--) {
            int c = 0;
            while(top!=-1 && heights[i] > st[top]) {
                top--;
                c++;
            }
            if(top!=-1) c++;
            ans[i] = c;
            st[++top] = heights[i];
        }
        return ans;
    }
}

File: 1944 number-of-visible-people-in-a-queue/solution.cpp
class Solution {
public:
    vector<int> canSeePersonsCount(vector<int>& heights) {
        int n = heights.size();
        vector<int> ans(n,0);
        stack<int> st;
        for(int i=n-1;i>=0;i--) {
            int c = 0;
            while(!st.empty() && heights[i] > st.top()) {
                st.pop();
                c++;
            }
            if(!st.empty()) c++;
            ans[i] = c;
            st.push(heights[i]);
        }
        return ans;
    }
};
